lib/actors/examples/fannkuchredux/fannkuchredux.nit

```diff
@@ -36,21 +36,14 @@ class FannkuchRedux
 	fun first_permutation(idx: Int) do
 		for i in [0..p.length[ do p[i] = i
 
-		var i = count.length - 1
-		while i > 0 do
+		for i in [0..(count.length - 1)].reverse_iterator do
 			var d = idx / fact[i]
 			count[i] = d
 			idx = idx % fact[i]
 
 			p.copy_to(0, i+1, pp, 0)
 
-			for j in [0..i] do
-				if j + d <= i then
-					p[j] = pp[j+d]
-				else
-					p[j] = pp[j+d-i-1]
-				end
-			end
+			for j in [0..i] do p[j] = if j + d <= i then pp[j+d] else pp[j+d-i-1]
 			i -= 1
```

Comment:
`i` is set at the next iteration of the `for` loop anyway... Or is there something I'm missing?